sale_subscription_financial_risk/models/account_invoice.py

```diff
@@ -0,0 +1,26 @@
+from odoo import models, fields, api
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    # Bool which determines if we need to execute the financial risk check
+    # This is set True only once during the write action when payment is rec'd
+    run_suspension_check = fields.Boolean(default=False)
+
+    @api.multi
+    def write(self, vals):
+        # If we are writing to 'state' to value 'paid', run the suspend check
+        if 'state' in vals:
```

Comment:
Two if can be combined

```suggestion
        if vals.get('state', False) == 'paid':
```

IMP else is not needed since run_suspension_check will be False by default

Reply:
Condensed second conditional check as recommended; very nice trick!